app/views/core/SubscribeModal.coffee

```diff
@@ -132,10 +134,19 @@ module.exports = class SubscribeModal extends ModalView
       @onSubscriptionError(jqxhr)
 
   startStripeSubscribe: ->
+    @startStripeSubscription(@basicProduct)
+
+  startYearlyStripeSubscription: ->
+    @startStripeSubscription(@basicProductYearly)
+
+  ###
+    Starts a stripe subscription based on the product passed in.
+  ###
+  startStripeSubscription: (product) ->
     application.tracker?.trackEvent 'Started subscription purchase', { service: 'stripe' }
     options = @stripeOptions {
-      description: $.i18n.t('subscribe.stripe_description')
-      amount: @basicProduct.adjustedPrice()
+      description: if product.get('name') is 'basic_subscription_yearly' then $.i18n.t('subscribe.stripe_yearly_description') else $.i18n.t('subscribe.stripe_description')
+      amount: product.adjustedPrice()
```

Comment:
These values are entirely aesthetic. Still need to ensure adjustedPrice visually reflects coupons.